presto-kafka/src/main/java/io/prestosql/plugin/kafka/KafkaFieldValueProvider.java

```diff
@@ -0,0 +1,172 @@
+
+/**
+ *
+ */
+public class KafkaFieldValueProvider
+        implements EncoderFieldValueProvider
+{
+    private final EncoderColumnHandle columnHandle;
+    private final Block block;
+    private final int position;
+    private final Type type;
+
+    public KafkaFieldValueProvider(
+            EncoderColumnHandle columnHandle,
+            Block block,
+            int position)
+    {
+        this.columnHandle = requireNonNull(columnHandle, "columnHandle is null");
+        this.block = requireNonNull(block, "block is null");
+        this.position = position;
+        this.type = columnHandle.getType();
+    }
+
+    @Override
+    public EncoderColumnHandle getColumn()
+    {
+        return columnHandle;
+    }
+
+    @Override
+    public boolean getBoolean()
+    {
+        return type.getBoolean(block, position);
+    }
+
+    @Override
+    public long getLong()
+    {
+        return type.getLong(block, position);
+    }
+
+    @Override
+    public int getInt()
+    {
+        return toIntExact(type.getLong(block, position));
+    }
+
+    @Override
+    public short getShort()
+    {
+        return Shorts.checkedCast(type.getLong(block, position));
+    }
+
+    @Override
+    public byte getByte()
+    {
+        return SignedBytes.checkedCast(type.getLong(block, position));
+    }
+
+    @Override
+    public double getDouble()
+    {
+        return type.getDouble(block, position);
+    }
+
+    @Override
+    public float getReal()
+    {
+        return intBitsToFloat(toIntExact(type.getLong(block, position)));
+    }
+
+    @Override
+    public String getVarchar()
+    {
+        return type.getSlice(block, position).toStringUtf8();
+    }
+
+    @Override
+    public String getChar()
+    {
+        return padSpaces(type.getSlice(block, position), ((CharType) type)).toStringUtf8();
+    }
+
+    @Override
+    public ByteBuffer getVarbinary()
+    {
+        return type.getSlice(block, position).toByteBuffer();
+    }
+
+    @Override
+    public Slice getSlice()
+    {
+        return Varchars.truncateToLength(type.getSlice(block, position), type);
+    }
+
+    @Override
+    public Block getBlock()
+    {
+        return (Block) type.getObject(block, position);
+    }
+
+    @Override
+    public boolean isNull()
+    {
+        return block.isNull(position);
+    }
+
+    @Override
+    public int hashCode()
+    {
+        return Objects.hash(columnHandle, block, position);
+    }
+
+    @Override
+    public boolean equals(Object obj)
+    {
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null || getClass() != obj.getClass()) {
+            return false;
+        }
+
+        KafkaFieldValueProvider other = (KafkaFieldValueProvider) obj;
+        return Objects.equals(this.columnHandle, other.columnHandle)
+                && Objects.equals(this.block, other.block)
+                && Objects.equals(this.position, other.position);
+    }
```

Comment:
does this class need equals?

block,position -based equality is not really value-based equality, so i suspect it does not

Reply:
no it doesn't, I never use it

Reply:
So please remove it.